client/state/selectors/get-sites.js

```diff
@@ -11,9 +17,10 @@ import { getSite } from 'state/sites/selectors';
  * @return {Array}        Sites objects
  */
 export default createSelector(
-	( state ) => (
-		Object.values( state.sites.items )
-			.map( site => getSite( state, site.ID ) )
-	),
+	( state ) => {
+		const primarySiteId = getPrimarySiteId( state );
+		const [ primarySite, sites ] = partition( state.sites.items, site => site.ID === primarySiteId );
```

Comment:
Side thought: I like lodash's shorthand for this: `partition( items, { ID: primarySiteId } )`.